deployment.go

```diff
@@ -72,7 +82,28 @@ func (r *marathonClient) Deployments() ([]*Deployment, error) {
 	if err != nil {
 		return nil, err
 	}
-
+	// Allows loading of deployment steps from the Marathon v1.X API
+	// Implements a fix for issue https://github.com/gambol99/go-marathon/issues/153
+	for _, deployment := range deployments {
+		// Unmarshal pre-v1.X step
+		if err := json.Unmarshal(deployment.XXStepsRaw, &deployment.Steps); err != nil {
+			deployment.Steps = make([][]*DeploymentStep, 0)
+			steps := make([]*DeploymentStepV1, 0)
```

Comment:
nit-pick: You can also use `var steps []*DeploymentStepV1` here to save one allocation.